### ISSUE TITLE
CORE::readlink inconsistent on some distros EINVAL rather than ENOENT (FreeBSD?)

### ISSUE THREAD

**Issue by toddr** [cpanel/Test-MockFile#84]
http://www.cpantesters.org/cpan/report/9a7224ce-7588-11ec-8317-697f1f24ea8f
http://www.cpantesters.org/cpan/report/90cfa0d6-7588-11ec-8317-697f1f24ea8f
http://www.cpantesters.org/cpan/report/94f628ba-7588-11ec-8317-697f1f24ea8f
http://www.cpantesters.org/cpan/report/8e1690d4-7588-11ec-8317-697f1f24ea8f
http://www.cpantesters.org/cpan/report/9d201adc-7588-11ec-8317-697f1f24ea8f

```
# Failed test '$! is ENOENT for a readlink(undef)'
# at t/readlink.t line 52.
# +-----+----+-------+
# | GOT | OP | CHECK |
# +-----+----+-------+
# | 22  | eq | 2     |
# +-----+----+-------+

# Failed test '$! is ENOENT for a readlink(undef)'
# at t/readlink.t line 58.
# +-----+----+-------+
# | GOT | OP | CHECK |
# +-----+----+-------+
# | 22  | eq | 2     |
# +-----+----+-------+
# Seeded srand with seed '20220114' from local date.
```

**Comment by toddr** [comment]
```
        EINVAL => 22,
```

**Comment by toddr** [comment]
I think this is yet another distro specific bug likely related to docker where the error emitted is not consistent with the norm.
```perl
$! = 0;
my $got = 'abc';
like( warning { $got = CORE::readlink(undef) }, qr/^Use of uninitialized value in readlink at /, "Got expected warning for passing no value to readlink" );
is( $got,   undef,  "readlink without args is undef." );
is( $! + 0, ENOENT, '$! is ENOENT for a readlink(undef)' );
```

**Comment by xsawyerx** [comment]
Resolved by #130 by supporting FreeBSD's `EINVAL` in cases of `readlink(under)`.
